go/saltpack/errors.go

```diff
@@ -47,6 +47,23 @@ var (
 
 	// ErrBadLookup is when the user-provided key lookup gives a bad value
 	ErrBadLookup = errors.New("bad key lookup")
+
+	// ErrNoStream is returned if a stream constructor returned a nil
+	// stream.
+	ErrNoStream = errors.New("no stream")
```

Comment:
Can you get read of this?  It seems like `newSignAttachedStream` should either return (non-nil, nil) or (nil, non-nil). I.e., we don't have to handle the (nil, nil) case.